lib/iex/lib/iex/evaluator.ex

```diff
@@ -10,33 +10,51 @@ defmodule IEx.Evaluator do
     * keeping expression history
 
   """
-  def start(server, leader) do
+  def start(server, leader, opts) do
     old_leader = Process.group_leader
     Process.group_leader(self, leader)
 
     try do
-      loop(server, IEx.History.init)
+      loop(server, IEx.History.init, run_state(opts))
     after
       Process.group_leader(self, old_leader)
     end
   end
 
-  defp loop(server, history) do
+  defp loop(server, history, evaluator_state) do
```

Comment:
Since it is evaluator's state we could name it just `state` and use `iex_state` for server's state respectively.
I think it'll reduce visual noise and improve readability.